projects/addon-mobile/directives/ripple/ripple.directive.ts

```diff
@@ -41,60 +35,108 @@ class TuiRippleStyles {}
-                        touchEnd$.pipe(map(TUI_TRUE_HANDLER)),
-                    ).pipe(
-                        take(1),
-                        // eslint-disable-next-line rxjs/no-unsafe-takeuntil
-                        takeUntil(touchMove$),
-                        tap(() => {
-                            this.renderer.setStyle(
-                                ripple,
-                                'background',
-                                this.tuiRipple || null,
-                            );
-                            this.renderer.appendChild(this.el, ripple);
-                        }),
-                        switchMap(isTap =>
-                            isTap
-                                ? animationEndOn$
-                                : race(
-                                      touchEnd$.pipe(switchMap(() => animationEndOn$)),
-                                      animationEndOn$.pipe(switchMap(() => touchEnd$)),
-                                  ),
+        race(
+            touchEnd$.pipe(switchMap(() => animationEndOn$)),
+            animationEndOn$.pipe(switchMap(() => touchEnd$)),
+        )
+            .pipe(
+                first(),
+                takeUntil(touchMove$),
+                defaultIfEmpty(undefined),
+                switchMap(() =>
+                    tuiTypedFromEvent(
+                        ripple.animate(
+                            [
+                                {
+                                    opacity: '0.12',
+                                },
+                                {
+                                    opacity: '0',
+                                },
+                            ],
+                            {
+                                duration: this.duration,
+                            },
                         ),
-                        map(() => ripple),
-                    );
-                }),
-                takeUntilDestroyed(),
+                        'finish',
+                    ),
+                ),
+                tap(() => element.removeChild(ripple)),
+                takeUntilDestroyed(this.destroyRef),
             )
-            .subscribe(element =>
-                this.renderer.setStyle(element, 'animationName', RIPPLE_OFF),
-            );
+            .subscribe();
+    }
+
+    private createRipple(
+        clientX: number,
+        clientY: number,
+        {width, height, top, left}: DOMRect,
+    ): HTMLElement {
+        const ripple: HTMLElement = this.document.createElement('div');
+        const radius = Math.sqrt(width * width + height * height);
+        const dimension = radius * 2;
+        const x = clientX - left - radius;
+        const y = clientY - top - radius;
+
+        ripple.classList.add('tui-ripple');
+        ripple.style.width = tuiPx(dimension);
+        ripple.style.height = tuiPx(dimension);
+        ripple.style.left = tuiPx(x);
+        ripple.style.top = tuiPx(y);
```

Comment:
Maybe we can just use `JSON.stringify` + [cssText](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/cssText) and get rid of less file completely, instead of adding class and keeping a style only component? The rules there are pretty simple, no need for compile processing/autoprefixr.